storage/remote/otlptranslator/prometheus/helpers_from_stdlib.go

```diff
@@ -67,7 +70,10 @@ func getTokensAndSeparators(s string, f func(rune) bool, allowUTF8 bool) ([]stri
 				separators = append(separators, s[tokens[i-1].end:tokens[i].start])
 			} else {
 				sep := ""
-				for j := tokens[i-1].end; j < tokens[i].start; j++ {
+				// We measure the rune count instead of using len because len returns amount of
+				// bytes in the string, and utf-8 runes can be multiple bytes long.
+				runeCount := utf8.RuneCountInString(s[tokens[i-1].end:tokens[i].start])
+				for j := 0; j < runeCount; j++ {
 					sep += "_"
```

Comment:
As far as my little knowledge of this codebase gets, this looks like quite a hot path, so I would avoid building a string like that (this is the least efficient way, probably).

Building a string using a `strings.Builder` is more efficient, and in this specific case you can use `strings.Repeat`.

However, in this case as we know that we're always repeating underscores I would do:

```go

const underscores = "_________" // 128 underscores, or whatever we think is the pessimistic case here

...

if runeCount <= len(underscores) {
    sep = underscores[:runeCount]
} else {
    sep = strings.Repeat("_", runeCount) 
}
```